datacube/index/hl.py

```diff
@@ -179,11 +182,16 @@ def resolve_ds(ds, sources, cache=None):
 
             doc = ds.doc
 
-            db_ds = db_dss.get(ds.id)
-            if db_ds:
-                product = db_ds.type
+            if verify_lineage:
```

Comment:
it's already True here

Reply:
I am reading it again, it still seems to me that it is going to be executed regardless.

Reply:
I don't like this approach, it increase number of DB queries in the default case. Ideally one would either run `bulk_get` OR run cheaper `bulk_has`, this approach does both - not good